### PR TITLE
Mods 29

### PR DESCRIPTION
**Pull request by gabrielbenavidez**
- Removed all of the op and sm flag reference.
- Parses lpass CMS naming syntax and converts into SM.

### CODE REVIEW

File: db_facts/db_info.py [hunk 1]
```diff
@@ -69,6 +71,7 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
                 dbcli_config['exports_from'][method]['pull_lastpass_from']
             lastpass_entry_name = template(template_for_lastpass_entry_name,
                                            (db_info, {}))
+            lastpass_entry_name = translate_secret_id_to_sm(lastpass_entry_name)
```
Comment (Brunope) [inline db_facts/db_info.py:74]: mentioned this to you in a DM but posting here so we don't forget - looks like this is converting ALL db-facts entries to secretsmanager ones, which isn't what we always want to do.

File: db_facts/db_info.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 from .jinja_context import pull_jinja_context
 from .errors import fail_on_invalid_db_name
 from .config import load_config
-from .lpass import pull_lastpass_username_password, db_info_from_lpass, pull_lastpass_aws_iam
+from .lpass import pull_lastpass_username_password, db_info_from_lpass, pull_lastpass_aws_iam, service_account_db_info_from_lpass
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:7]: [flake8] E501 line too long (129 > 100 characters)

File: db_facts/db_info.py [hunk 3]
```diff
@@ -69,7 +69,24 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
                 dbcli_config['exports_from'][method]['pull_lastpass_from']
             lastpass_entry_name = template(template_for_lastpass_entry_name,
                                            (db_info, {}))
-            additional_attributes = \
+
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:72]: [flake8] W293 blank line contains whitespace

File: db_facts/db_info.py [hunk 4]
```diff
@@ -69,7 +69,24 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
                 dbcli_config['exports_from'][method]['pull_lastpass_from']
             lastpass_entry_name = template(template_for_lastpass_entry_name,
                                            (db_info, {}))
-            additional_attributes = \
+            
+            #if db is of type cms then convert secret names and keys accordingly.
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:73]: [flake8] E265 block comment should start with '# '

File: db_facts/db_info.py [hunk 5]
```diff
@@ -69,7 +69,24 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
...
+            if config['jinja_context_name'] == 'cms':
+                stack = jinja_context[0]['cms_env']
+                user = jinja_context[0]['cms_user']
+                service_accounts = ['list_cutter', 'etl_user', 'dbadmin', 'archive', 'monitor', 'syncmon']
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:77]: [flake8] E501 line too long (106 > 100 characters)

File: db_facts/db_info.py [hunk 6]
```diff
@@ -69,7 +69,24 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
...
+                if user in service_accounts:
+                    lastpass_entry_name = f'{stack}_vertica_service_accounts'
+                    additional_attributes = \
+                        service_account_db_info_from_lpass(lastpass_entry_name, user)
+                else:
+                    lastpass_entry_name = f'{stack}_vertica_{user}_creds'
+                    additional_attributes = \
+                    db_info_from_lpass(lastpass_entry_name)
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:85]: [flake8] E122 continuation line missing indentation or outdented

File: db_facts/db_info.py [hunk 7]
```diff
@@ -69,7 +69,24 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
...
+
+            #if db is not of type cms then skip last block
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:87]: [flake8] E265 block comment should start with '# '

File: db_facts/lpass.py [hunk 8]
```diff
@@ -52,3 +52,28 @@ def db_info_from_lpass(lpass_entry_name: str):
             'protocol': db_protocol(db_type),
             'port': port,
             'database': dbname}
+
+"""
```
Comment (bluelabsbutler) [inline db_facts/lpass.py:56]: [flake8] E305 expected 2 blank lines after class or function definition, found 1

File: db_facts/lpass.py [hunk 9]
```diff
@@ -52,3 +52,28 @@ def db_info_from_lpass(lpass_entry_name: str):
...
+Takes in a lastpass entry name and username and retrieves the db values
+from the password manager. This method is specifically used for
```
Comment (bluelabsbutler) [inline db_facts/lpass.py:58]: [flake8] W291 trailing whitespace

File: db_facts/lpass.py [hunk 10]
```diff
@@ -52,3 +52,28 @@ def db_info_from_lpass(lpass_entry_name: str):
...
+
+"""
+Takes in a lastpass entry name and username and retrieves the db values
+from the password manager. This method is specifically used for 
+the service manager accounts as the keys syntax are different.
+    Parameters:
+        lpass_entry_name (str): the service account name in secretsmanager
+        user (str): the user of the service account
+"""
+def service_account_db_info_from_lpass(lpass_entry_name: str, user:str):
```
Comment (bluelabsbutler) [inline db_facts/lpass.py:64]: [flake8] E302 expected 2 blank lines, found 0

Comment (bluelabsbutler) [inline db_facts/lpass.py:64]: [flake8] E231 missing whitespace after '

File: db_facts/db_info.py [hunk 11]
```diff
@@ -4,7 +4,10 @@
 from .jinja_context import pull_jinja_context
 from .errors import fail_on_invalid_db_name
 from .config import load_config
-from .lpass import pull_lastpass_username_password, db_info_from_lpass, pull_lastpass_aws_iam
+from .lpass import (
+    pull_lastpass_username_password, db_info_from_lpass,
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:8]: [flake8] W291 trailing whitespace

File: db_facts/db_info.py [hunk 12]
```diff
@@ -69,7 +72,23 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
                 dbcli_config['exports_from'][method]['pull_lastpass_from']
             lastpass_entry_name = template(template_for_lastpass_entry_name,
                                            (db_info, {}))
-            additional_attributes = \
+            # if db is of type cms then convert secret names and keys accordingly.
+            if config['jinja_context_name'] == 'cms':
+                stack = jinja_context[0]['cms_env']
+                user = jinja_context[0]['cms_user']
+                service_accounts = ['list_cutter', 'etl_user', 'dbadmin',
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:79]: [flake8] W291 trailing whitespace

File: db_facts/db_info.py [hunk 13]
```diff
@@ -69,7 +72,23 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
...
+            # if db is of type cms then convert secret names and keys accordingly.
+            if config['jinja_context_name'] == 'cms':
+                stack = jinja_context[0]['cms_env']
+                user = jinja_context[0]['cms_user']
+                service_accounts = ['list_cutter', 'etl_user', 'dbadmin', 
+                                    'archive', 'monitor', 'syncmon']
+                if user in service_accounts:
+                    lastpass_entry_name = f'{stack}_vertica_service_accounts'
+                    additional_attributes = \
+                    service_account_db_info_from_lpass(lastpass_entry_name, user)
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:84]: [flake8] E122 continuation line missing indentation or outdented

File: db_facts/lpass.py [hunk 14]
```diff
@@ -52,3 +52,29 @@ def db_info_from_lpass(lpass_entry_name: str):
...
+    db_protocol = 'vertica'
+    dbname = lpass_field(lpass_entry_name, 'database')
+
+    return {'password': password,
+            'host': host,
+            'user': user,
+            'type': db_type,
+            'protocol': db_protocol,
+            'port': port,
+            'database': dbname}
```
Comment (bluelabsbutler) [inline db_facts/lpass.py:80]: [flake8] W292 no newline at end of file

**Pull request closed without merging**